src/status_im2/common/scan_qr_code/view.cljs

```diff
@@ -0,0 +1,239 @@
+(ns status-im2.common.scan-qr-code.view
+  (:require [clojure.string :as string]
+            [reagent.core :as reagent]
+            [oops.core :as oops]            
+            [status-im2.common.scan-qr-code.style :as style]
+            [react-native.camera-kit :as camera-kit]
+            [re-frame.core :as re-frame]
+            [status-im.multiaccounts.key-storage.core :as key-storage]
+            [react-native.core :as rn]
+            [quo2.core :as quo]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
+            [react-native.blur :as blur]
+            [react-native.hole-view :as hole-view]
+            [quo2.foundations.colors :as colors]
+            [status-im.utils.platform :as platform]
+            [react-native.permissions :as permissions]
+            [status-im2.contexts.syncing.utils :as sync-utils]
+            [status-im2.contexts.onboarding.common.background.view :as background]
+            [react-native.safe-area :as safe-area]
+            ))
+
+;; Android allow local network access by default. So, we need this check on iOS only.
+(defonce preflight-check-passed? (reagent/atom (if platform/ios? false true)))            
```

Comment:
why is it an atom?

Reply:
Why are we recreating the same global atoms in multiple places?
Can we centralise these somewhere? cc @smohamedjavid , @siddarthkay

Reply:
Yes, we should. It's not good to have copies of the same atom again.

We can have a global atom to store all permissions. Should `react-native.permissions` ns work?

Reply:
shouldn't we use re-frame for this?

Reply:
> shouldn't we use re-frame app-db for this?

That works too.

Reply:
yep both sound good. Can we create an issue for this with a proposed solution?

Reply:
issue : https://github.com/status-im/status-mobile/issues/16354 , will add more context in a bit.

Reply:
> shouldn't we use re-frame app-db for this?

Yes, let's avoid global ratoms if we can and rely on the re-frame architecture. The codebase is littered with unnecessary global ratoms.

Reply:
so my question was more about dynamic, it seems like a constant